### ISSUE TITLE
Version of macOS Sierra

### ISSUE THREAD

**Issue by dieterdemeyer** [bikkelbroeders/TouchBarDemoApp#1]
Hello,

Is there a big difference between macOS Sierra 10.12.1 build 16B2657 and 16B2555 for this app to work ?
Because I have the latest version through the App Store, which is version 16B2555 and released on October 25h. When starting TouchBarServer, I get an error stating that I have the wrong version for this to work..

But according to the link you provided, that update dates from October 27th.
However, both refer to the same set of changes found at https://support.apple.com/kb/HT206174.

Thanks.

**Comment by robbertkl** [comment]
Hi,

You do require build 16B2657, which adds the SDK for controlling/interacting with the Touch Bar. The error message you're seeing confirms the app could not find the necessary framework.

Actually, recent upgrades to Sierra (a Mac I've upgraded yesterday) already end up with 16B2657, so it seems Apple has begun distributing this build. However, since you've probably upgraded to Sierra before this, you either need to download the update through the [link provided in the README](https://support.apple.com/kb/dl1897), or wait until the next regular Sierra update (10.12.2).

Hope this clarifies it. Good luck!

**Comment by dieterdemeyer** [comment]
Hello,

Indeed, the TouchBarServer app has no problem starting with build 16B2657 of macOS Sierra.
I suppose there are no side effects with installing macOS updates outside of the App Store ?

Could this also work through a wireless connection ?
Currently, for this to work, you need to connect your iOS device through a USB cable.

Thanks in advance.

**Comment by robbertkl** [comment]
No, there should be no side effects, since this is not a Beta or Developer Preview, but a regular version. In fact, everyone who's upgrading to Sierra right now will get this same version. You will still get 10.12.2 as a normal update through the App Store.

As for WiFi, we've actually tried this as well. While it worked, it did not perform as well as the USB connection. The frame rate of the animations and responsiveness to input were both lacking, so we decided to just stick with USB for the time being. We might add an option for WiFi at a later moment.
